src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,63 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        // Initialize a StringBuilder to store the report
+        StringBuilder stringBuilder = new StringBuilder();
+
+        // Create a DateTimeFormatter to parse the date strings
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
+        // Parse the dateFrom and dateTo strings into LocalDate objects
+        LocalDate fromDate = LocalDate.parse(dateFrom, formatter);
+        LocalDate toDate = LocalDate.parse(dateTo, formatter);
+
+        // Initialize an array to hold the LocalDate objects for each day in the data array
+        LocalDate[] dataArr = new LocalDate[data.length];
+
+        // Convert the dates in the data array into LocalDate objects
+        // and store them in the dataArr array
+        for (int i = 0; i < data.length; i++) {
+            dataArr[i] = LocalDate.parse(data[i].substring(0, 10), formatter);
```

Comment:
Why do you need this `substring`?

Reply:
"25.04.2019 John 60 50" input string, formatter need pattern first 10 symbols, so i use substring